plugins/BEdita/Core/src/Utility/SchemaTools.php

```diff
@@ -0,0 +1,83 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2024 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace BEdita\Core\Utility;
+
+use Cake\Database\Schema\TableSchemaInterface;
+
+class SchemaTools
+{
+    /**
+     * Get primary fields from schema.
+     *
+     * @param \Cake\Database\Schema\TableSchemaInterface $schema Table schema.
+     * @param array $options Options.
+     * @return array
+     */
+    public static function getPrimaryFields(TableSchemaInterface $schema, array $options = []): array
+    {
+        $fields = [];
+        foreach ($schema->constraints() as $name) {
+            $constraint = $schema->getConstraint($name);
+            if ($constraint['type'] === 'primary') {
+                if (empty($options['count']) || count($constraint['columns']) === $options['count']) {
+                    $fields = array_merge($fields, $constraint['columns']);
+                }
+            }
```

Comment:
Minor but we can simplify a little removing nested `if`

```suggestion
            if ($constraint['type'] !== 'primary') {
                continue;
            }
            
            if (empty($options['count']) || count($constraint['columns']) === $options['count']) {
                $fields = array_merge($fields, $constraint['columns']);
            }
```